pkg/generators/templates/naming_test.go

```diff
@@ -0,0 +1,15 @@
+package templates
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestTypeDisplayName(t *testing.T) {
+	require.Equal(t, "NullableString", TypeDisplayName("*string"))
+	require.Equal(t, "SliceString", TypeDisplayName("[]string"))
```

Comment:
I think this and everything below should be inverted as `StringSlice`, `InterfaceSlice`, `InterfaceSliceSlice` and `NullableInterfaceSliceSlice`.

Reply:
I inverted it now and added a new tests case to demonstrate it in a model